serrano/resources/field/base.py

```diff
@@ -82,7 +85,14 @@ def get_link_templates(self, request):
         }
 
     def get_queryset(self, request, params):
-        queryset = self.model.objects.all()
+        tree = trees[params['tree']]
+        q = Q()
+
+        # No public method for accessing the local models on the tree
+        for app_name, model_name in tree._models:
+            q |= Q(app_name=app_name, model_name=model_name)
+
+        queryset = self.model.objects.filter(q)
```

Comment:
Right, my bad. Didn't read that line clearly enough the first time. These
changes look good. Should be really useful.

-Don
On Jan 18, 2015 6:58 PM, "Byron Ruth" <notifications@github.com> wrote:

> In serrano/resources/field/base.py
> <https://github.com/chop-dbhi/serrano/pull/258#discussion-diff-23140993>:
>
> > @@ -82,7 +85,14 @@ def get_link_templates(self, request):
> >          }
> >
> >      def get_queryset(self, request, params):
> > -        queryset = self.model.objects.all()
> > +        tree = trees[params['tree']]
> > +        q = Q()
> > +
> > +        # No public method for accessing the local models on the tree
> > +        for app_name, model_name in tree._models:
> > +            q |= Q(app_name=app_name, model_name=model_name)
> > +
> > +        queryset = self.model.objects.filter(q)
>
> Doing an OR on the DataConcept filtering requires only one field to match
> the condition. A concept could be created that includes a field from an
> excluded or unrelated model, so the requirement is to exclude concepts if
> any fields do not match.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chop-dbhi/serrano/pull/258/files#r23140993>.
>